source/daplink/interface/target_flash.c

```diff
@@ -115,19 +115,30 @@ static error_t target_flash_program_page(uint32_t addr, const uint8_t *buf, uint
 
         if (config_get_automation_allowed()) {
             // Verify data flashed if in automation mode
-            while (write_size > 0) {
-                uint8_t rb_buf[16];
-                uint32_t verify_size = MIN(write_size, sizeof(rb_buf));
-                if (!swd_read_memory(addr, rb_buf, verify_size)) {
-                    return ERROR_ALGO_DATA_SEQ;
-                }
-                if (memcmp(buf, rb_buf, verify_size) != 0) {
+            if (flash->verify != 0) {
+                if (!swd_flash_syscall_exec(&flash->sys_call_s,
+                                    flash->verify,
+                                    addr,
+                                    flash->program_buffer_size,
+                                    flash->program_buffer,
+                                    0)) {
                     return ERROR_WRITE;
                 }
-                addr += verify_size;
```

Comment:
I suggest to add                 
```
addr += write_size;
buf += write_size;
size -= write_size;
```
Otherwaise it will crash when succefull virefied during flashing the image.